### PR TITLE
[move] Updated Vector functions

### PR DESCRIPTION
**Pull request by tnowacki**
##  Motivation

- Taking another look at the Vector library functions
- Removed get/set. borrow/borrow_mut can be used instead
- Added O(n) remove function
- Reimplemented remove_unstable. Renamed to swap_remove
  - It serves as a nice O(1) analogous function to remove

## Test Plan

- wrote tests

### CODE REVIEW

File: language/move-lang/src/lib.rs [hunk 1]
```diff
@@ -132,7 +132,8 @@ pub fn output_compiled_units(
         })
         .collect::<io::Result<Vec<_>>>()?;
     for (mut file, compiled_unit) in files_and_units {
-        file.write_all(&compiled_unit.serialize())?;
+        file.write_all(format!("{:#?}", compiled_unit).as_bytes())?;
+        // file.write_all(&compiled_unit.serialize())?;
```
Comment (sblackshear) [inline language/move-lang/src/lib.rs]: Delete?

File: language/move-lang/stdlib/modules/vector.move [hunk 2]
```diff
@@ -56,4 +55,35 @@ module Vector {
...
+
+        len = len - 1;
+        while (i < len) swap(v, i, { i = i + 1; i });
+        pop_back(v)
+    }
+
+    // Remove the `i`th element E of the vector by swapping it with the last element,
+    // and then popping it off
+    // It is O(1), but does not preserve ordering
+    public swap_remove<Element>(v: &mut T<Element>, i: u64): Element {
```
Comment (sblackshear) [inline language/move-lang/stdlib/modules/vector.move]: Looks much nicer in the source lang!

Will the borrow checker complain if you try to inline `last_idx` here?

File: language/move-lang/stdlib/modules/vector.move [hunk 3]
```diff
@@ -56,4 +55,35 @@ module Vector {
...
+        };
+        false
+    }
+
+    // Remove the `i`th element E of the vector, shifting all subsequent elements
+    // It is O(n) and preserves ordering
+    public remove<Element>(v: &mut T<Element>, i: u64): Element {
+        let len = length(v);
+        // i out of bounds; abort
+        if (i >= len) abort 10;
```
Comment (sblackshear) [inline language/move-lang/stdlib/modules/vector.move]: We should probably pick a convention for the VECTOR_INDEX_OUT_OF_BOUNDS error code + add it to vm/src/errors.rs at some point.

File: language/move-lang/src/lib.rs (hunk 1 above)
Comment (tnowacki) [reply language/move-lang/src/lib.rs]: oops

File: language/move-lang/stdlib/modules/vector.move (hunk 2 above)
Comment (tnowacki) [reply language/move-lang/stdlib/modules/vector.move]: It's a bit weird, and I'll need to talk to @shazqadeer about it. In short, I think we are doing some non optimal things with regards to borrow local and copy local

**Pull request closed without merging**